src/policy.c

```diff
@@ -142,6 +142,16 @@ device_matches_rule(rule_t *rule, device_t *device)
   if (rule->dev_deviceid != 0 &&
       device->deviceid != rule->dev_deviceid)
     return false;
+  /* If the rule specifies a serial, it has to match */
+  if (rule->dev_serial != NULL)
+  {
+    /* If the device does not have a serial, it does not match */
+    if (device->serial == NULL)
+      return false;
+    /* Ensure the rule and device serials match */
+    else if (strcmp(rule->dev_serial,device->serial) != 0)
+      return false;
+  }
```

Comment:
Modified this portion to include the check for when the rule states there should be a serial but the device does not have/present a serial